### DISCUSSION TITLE
Ned NES dumping help

### DISCUSSION THREAD

**Discussion by mew4900** [sanni/cartreader#348] [Need help]
Forgive my ignorance, but many of my nes cartridges are outputting as .bin. Sometimes I get a .nes and those run fine, but it’s only happened for certain carts. Can someone please help me to make these nes .bin files useful?  Thanks!

**Comment by sanni** [comment]
The difference between bin and nes file is the 16 byte emulation header. This is not found in the cartridge. The Cart Reader has a database with all the emulation headers and tries to add the header to the bin file after dumping but it might not recognize all the games.

To manually add the emulation header you can go to http://nes.dnsabr.com and search for your game. Once found copy the 32 letters/numbers from the Header (hex) column and paste them in front of the data in the nes bin file using a hex editor.

![emuheader0](https://user-images.githubusercontent.com/19169494/158785881-758cef89-49e6-4e77-be5d-d41ffe2d9e3a.jpg)

![emuheader1](https://user-images.githubusercontent.com/19169494/158785599-8be636a6-325e-4a52-b284-c011568fb857.jpg)

**Comment by mew4900** [comment]
Thanks so much!  It’s working now. I’m super-thankful that this community is here and nice to even newbies like me. Thanks again.

**Comment by Androxilogin** [comment]
If you get a .bin can you list a way of checking them on the NES page? Also, if the title is not found is there a way we can add it to the list for merging info?

**Comment by sanni** [comment]
You can identify a cartridge either with the help of the combined rom CRC32 or the individual CRC32s of the PRG and CHR files. If you got 7zip installed there is an option for CRC32 in the right-click menu.

![header0](https://user-images.githubusercontent.com/19169494/158988990-3a936e83-3490-4b11-97cb-7bc6c804d3ae.jpg)

![header1](https://user-images.githubusercontent.com/19169494/158988214-324d8b70-fbf9-4720-bd78-60d311246ec1.jpg)

![header2](https://user-images.githubusercontent.com/19169494/158985830-21c375f9-a291-431a-87c8-90a110cdf7b7.jpg)

![header3](https://user-images.githubusercontent.com/19169494/158985825-21a5a88c-9631-40e4-96f1-a73a1a52f08a.jpg)

**Reply by Androxilogin** [reply]
One last question, why does this happen? For example, I have dumped Bayou Billy multiple times in the past and ended up discarding the files because I assumed they were just bad dumps. Is it just because they aren't in the database? I'd like to contribute what I acquire along the way if they aren't in the NES file, I'm just not sure how to go about doing this.

**Reply by sanni** [reply]
Most likely a bad dump or the Arduino did an error when calculating the CRC32 checksum. 
In such a case you can copy the PRG/CHR/BIN ROM files to your PC and calculate their CRC32 checksums there again and then search for them in one of the (online) databases to verify if it's a good dump.
